### ISSUE TITLE
Generator : Discriminator updates

### ISSUE THREAD

**Issue by koutilya-pnvr** [dmerrick520/Learning-from-Simulated-and-Unsupervised-Images-through-Adversarial-Training-SimGAN-PyTorch#2]
In the original paper, the generator is updated 50 times (Kg=50) for an update of the discriminator (Kd=1), but in your repo you were training with Kg=2 and Kd=1. Just confirming if you were in same page as the paper.

**Comment by daniel-merrick** [comment]
I could not get good results with kg =50. The refiner was collapsing (mode collapse) with that high of kg. In apples blog https://machinelearning.apple.com/2017/07/07/GAN.html they claimed they use a high kg because the discriminator converges faster due to batch norm. But in the actual paper they don't mention using batch norm, I don't use batch-norm and therefore I guess I don't need a high kg.

**Comment by koutilya-pnvr** [comment]
Hmm. Bummer. Its true that the original paper doesn't use BatchNorm any where despite having Kg=50 and was able to achieve convergence somehow. Any thoughts on that?

On a different note, you were using BatchNorm after the very first conv layer in the Refiner (before the resnet blocks) Any reason for that?
https://github.com/dmerrick520/Learning-from-Simulated-and-Unsupervised-Images-through-Adversarial-Training-SimGAN-PyTorch/blob/e70fbb89e11896a112eb4ff854dabcf73d597413/simgan/lib/simgan_network.py#L45

**Comment by daniel-merrick** [comment]
Perhaps they forgot to mention they used batchnorm, not certain. They also don't mention any use of activation functions. Not really sure why they would leave this information out. 

And that was probably a mistake. I was doing a lot of experimenting. Almost certain the results shown are not with the batchnorm, I will update. 

Thanks

**Comment by koutilya-pnvr** [comment]
Great, thanks for sharing your experience. Also, I was mailing you (dmerrick@purdue.edu) regarding some of my thoughts on this work. Hope you find them helpful.
